river/tree/_nodes/hatc_nodes.py

```diff
@@ -287,8 +293,13 @@ def learn_one(self, x, y, *, sample_weight=1., tree=None, parent=None, parent_br
 
     def predict_one(self, x, *, tree=None):
         # In case split nodes end up being used (if emerging categorical feature appears,
-        # for instance)
-        return self.stats  # Use the MC (majority class) prediction strategy
+        # for instance) use the MC (majority class) prediction strategy
+        dist = {c: 0. for c in self.stats}
+        votes_sum = sum(self.stats.values())
+        if votes_sum > 0:
```

Comment:
We can take the chance to improve this method. It is not optimal to protect it externally and resulted in multiple issues opened in the past due to the error throwing

```python
def normalize_values_in_dict(dictionary, factor=None, inplace=True, raise_error=False):
    """Normalize the values in a dictionary using the given factor.

    For each element in the dictionary, applies `value/factor`.

    Parameters
    ----------
    dictionary
        Dictionary to normalize.
    factor
        Normalization factor value. If not set, use the sum of values.
    inplace
        If True, perform operation in-place
    raise_error
        If `True`, will raise an error if the normalization factor is `0` or `None`. If `False`,
        will return gracefully if the normalization factor is `0` or `None`.

    """
    if factor is None:
        factor = sum(dictionary.values())
    if raise_error and (factor == 0 or math.isnan(factor)):
        raise ValueError(f'Can not normalize, normalization factor is {factor}')

    if not inplace:
        dictionary = copy.deepcopy(dictionary)

    if math.isnan(factor) or factor == 0:
        # Can not normalize, return gracefully
        return dictionary

    for key, value in dictionary.items():  # loop over the keys, values in the dictionary
        dictionary[key] = value / factor

    return dictionary
```

Reply:
You are totally right! And thank you so much for this comment. It helped spot some bugs in the Hoeffding Adaptive Tree that were not covered by our tests. I also updated this function as requested in commit [f2c6fad](https://github.com/creme-ml/creme/pull/376/commits/f2c6fad7fee9fa382de554c78bb20aa09a1e5033).